js/components/plates.js

```diff
@@ -266,11 +267,16 @@ export default class Plates extends PureComponent {
     } else if (data.stepIdx === 0 && snapshotAvailable) {
       this.setState({snapshotAvailable: false})
     }
-    this.setDensities(this.convertPlatesToDensities(data.plates))
-    this.setState({plateColors: data.plates.reduce(function (plateIdToColor, plate) {
-      plateIdToColor[plate.id] = plate.baseColor
-      return plateIdToColor
-    }, {})})
+    // Plate densities and colors are initialized by the model, but never changed by it
+    if (this.isStateObjectEmpty('plateDensities')) {
```

Comment:
This doesn't sound right to me. It's an assumption that might bite us in the future. And I'm pretty sure it will fail when you use authoring mode, pick one preset, go next, then go back, and pick another preset. Densities and colors won't be updated.

Reply:
I guess what we should do here is to check equality instead.